src/components/SessionExpiry.jsx

```diff
@@ -0,0 +1,64 @@
+import React, {useState, useEffect} from "react";
+
+const SessionExpiry = () => {
+  const [timeLeft, setTimeLeft] = useState(600);
+  const [isReset, setIsReset] = useState(false)
+  const [isLoggedIn, setIsLoggedIn] = useState(true)
+  const [isDismissed, setIsDismissed] = useState(false)
+
+  const dismiss = () => {
+    setIsDismissed(true)
+  }
+
+  const displayTime = (timeLeft) => {
+    const hh = Math.floor(timeLeft/60);
+    const mm = `${Math.floor(timeLeft % 60)}`.padStart(2, "0");   
+    return `${hh}:${mm}`;
+  }
+
+  const autoLogOut = () => {
+    const countDown = () => {
+      isReset ? setTimeLeft(600) && setIsReset(!isReset) : setTimeLeft(timeLeft-.1)
+    }    
+    if(isLoggedIn && (timeLeft > 0)){
```

Comment:
we discussed about user personalization not in the works for this release. is  `isLoggedIn` something diff/ ?
the condition is: if it has been not 1 hr and location == last_location;  return cache;
You will also get location from Search.jsx ; it is city location;
so if it has not been 1 hr and there is timeLeft for that and the user has not moved from the last location, then return already existing result else clear Cache;

Reply:
Ok.  I am confused.  I thought that the story was for user to be logged out after 10 minutes.  I will have to refactor for this accordingly.

Reply:
Initially it was. But In case of public/Anonymous usage there is no User.  I ll append to current story to account for Anonymous User. 
Add a `TODO(<ur_github>_id): ` on top of this method as a reminder so that LoggedIn feature can be added later.
Regardless of whether there is a signedUser or AnonymousUser, these conditions above :point_up: must hold true.

As for when we implement signedUser, we will clear the cache as well as logout the user and reload the page to default main view.

Reply:
Ok.  Just so I'm clear.  This Modal will pop up if the User is logged and and is about to be logged out.  It will also refresh the cache whether or not the user is logged in after a certain amount of time. And It will pop up a let the user know that they have been logged out.  Should it also pop up when the cache has been cleared?  A countdown for when the cache will be cleared?  Seems like this might be a separate modal?

Reply:
@ksr89 pls help out